.github/workflows/build.yml

```diff
@@ -21,29 +21,53 @@ jobs:
     runs-on: ubuntu-20.04
     outputs:
       build-doc: ${{ steps.set-matrix.outputs.build-doc }}
+      boards-aarch: ${{ steps.set-matrix.outputs.boards-aarch }}
       boards-arm: ${{ steps.set-matrix.outputs.boards-arm }}
-      boards-riscv: ${{ steps.set-matrix.outputs.boards-riscv }}
       boards-espressif: ${{ steps.set-matrix.outputs.boards-espressif }}
-      boards-aarch: ${{ steps.set-matrix.outputs.boards-aarch }}
+      boards-riscv: ${{ steps.set-matrix.outputs.boards-riscv }}
+      cp-version: ${{ steps.cp-version.outputs.cp-version }}
     steps:
     - name: Dump GitHub context
       run: echo "$GITHUB_CONTEXT"
       env:
         GITHUB_CONTEXT: ${{ toJson(github) }}
-    - uses: actions/checkout@v3
+    - name: Set up repository
+      uses: actions/checkout@v3
       with:
         submodules: false
         fetch-depth: 1
-    - name: Set up Python 3
+    - name: Set up python
       uses: actions/setup-python@v4
       with:
         python-version: "3.x"
-    - name: Get CP deps
-      run: python tools/ci_fetch_deps.py test ${{ github.sha }}
+    - name: Duplicate USB VID/PID check
+      run: python3 -u -m tools.ci_check_duplicate_usb_vid_pid
+    - name: Create submodule status
+      run: git submodule status extmod/ulab lib/ tools/ >> submodule_status
+    - name: Cache submodules
+      uses: actions/cache@v3
+      with:
+        path: |
+          .git/modules/
+          extmod/ulab
+          lib/
+          tools/
```

Comment:
This is repeated a bunch of places. Is it possible to factor it out into a definition at the top of the file or something like that, in case we want to change the list of cached submodules?

Reply:
I am not sure how to achieve this. An environment variable seems like an obvious solution and likely the only one but it would require two env variables as `path` doesn't seem to take a sequence or a string of space delimited paths.